### ISSUE TITLE
Geolocation

### ISSUE THREAD

**Issue by jmtellez** [jmtellez/CLI-mate#3]
When no command is passed, `cli-mate` should return weather data from where the user is located.

**Comment by CarlyRaeJepsenStan** [comment]
If you're using a bash script, you could try something like:
```
ipaddr=curl ipecho.net/plain ; echo
curl https://freegeoip.app/xml/$ipaddr
```
I can help!

**Comment by jmtellez** [comment]
you can make the request with node. LMK and I can assign it to you.

**Comment by CarlyRaeJepsenStan** [comment]
yes, we can use `node-fetch` instead to gather geolocation information - freegeoip.app returns longitude and latitude.

**Comment by jmtellez** [comment]
Use `postman-request`

**Comment by CarlyRaeJepsenStan** [comment]
Thanks.  Please send a pull request when you can.

**Comment by jmtellez** [comment]
Fork the project and then open PR when you are done!

**Comment by CarlyRaeJepsenStan** [comment]
@jmtellez How do I link the fork to a PR? 😅 Sorry, I've never worked with pull requests before.

**Comment by jmtellez** [comment]
Push your changes to your forked repo, then head over to the original repo and click on Pull request tab and create a PR. Select the branch you are wanting to merge your code to and add me as a reviewer. 

Here is a detailed doc on how to do it: https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github

**Comment by CarlyRaeJepsenStan** [comment]
Thanks!  I noticed this in the code:
```
switch (location) {
  case undefined:
```

How would we differentiate between showing the menu when the location is undefined, and automatically getting the IP address?

**Comment by jmtellez** [comment]
What you can do is create a function like utils/geocode.js and instead of hitting map box endpoint hit freegeoip endpoint. We can call that function and still get lat and Lon then call forecast()

**Comment by CarlyRaeJepsenStan** [comment]
freegeoip only returns city - I'll see if I can get the city out of freegeoip, and then hopefully pass that back into the other endpoints.

**Comment by jmtellez** [comment]
You can pass city to forecast() it should work

**Comment by CarlyRaeJepsenStan** [comment]
@jmtellez the function works for now - I'll try to pass the city in tomorrow.  

#8

**Comment by arpitvaghela** [comment]
location can be an optional argument if we are implementing this

**Comment by CarlyRaeJepsenStan** [comment]
@arpitvaghela that's handled with the switch statements.
@jmtellez Did you see #8 ?

**Comment by arpitvaghela** [comment]
I am trying to solve the #11, will change my implementation to support this as well

**Comment by cosmoimai** [comment]
Sir we just have to use geolocation for that. Is it?

**Comment by cosmoimai** [comment]
sir, we have to provide your own token? because it is not working means that process.env.MAPBOX is not working 
in geocode line 7

**Comment by CarlyRaeJepsenStan** [comment]
@cosmoimai did you generate your own api keys?
Also @arpitvaghela how should we implement the automatic geolocation with the unit flags?

**Comment by cosmoimai** [comment]
@CarlyRaeJepsenStan Sir I think we have to generate our own api key or some sort of token i think

**Comment by CarlyRaeJepsenStan** [comment]
@cosmoimai what kind of errors are you receiving?

**Comment by jmtellez** [comment]
@cosmoimai you need to create both weatherstack and mapbox api keys and set their corresponding env vars.

**Comment by cosmoimai** [comment]
@CarlyRaeJepsenStan sir, actually it is giving unable to locate

**Comment by CarlyRaeJepsenStan** [comment]
Using the autolocation: 
```
/* 
== Autolocation - should be fired on case undefined

const autoLocate = require("./utils/autolocate")


autoLocate((err, city) => {
 //console.log(city) ;

case geocode, etc.
})

==
*/
```
Placed here for future reference.

**Comment by CarlyRaeJepsenStan** [comment]
#22
